conan_provider.cmake

```diff
@@ -171,7 +171,7 @@ function(detect_lib_cxx LIB_CXX)
         set(${LIB_CXX} "libstdc++${_CONAN_GNU_LIBSTDCXX_SUFFIX}" PARENT_SCOPE)
     elseif(CMAKE_CXX_COMPILER_ID MATCHES "AppleClang")
         set(${LIB_CXX} "libc++" PARENT_SCOPE)
-    elseif(CMAKE_CXX_COMPILER_ID MATCHES "Clang" AND NOT CMAKE_SYSTEM_NAME MATCHES "Windows")
+    elseif(CMAKE_CXX_COMPILER_ID MATCHES "Clang" AND NOT MSVC)
```

Comment:
For the latter, yes:
> `MSVC`: Set to true when the compiler is some version of Microsoft Visual C++ **or another compiler simulating the Visual C++ cl command-line syntax.**

For an external Clang on Windows, I can only assume that it's set to false, but I have not tested this.

Reply:
I have just checked. Even for LLVM/Clang, that doesn't use ``libcxx``, but MSVC runtime, ``MSVC`` variable is not defined.
It seems that it would be good to improve the condition somehow, so that detection doesn't run for LLVM/Clang (MSVC backend)